### PR TITLE
Migrate prod-devel to Yocto-thud.

### PR DESCRIPTION
**Pull request by iartemenko**

### CODE REVIEW

**Comment by rshym** [comment]
I refrain any comments due to lack of understanding in this area.

**Comment by iartemenko** [comment]
Tested on Salvator-x-h3-4x2g

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xt-log/xt-log_git.bb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+DESCRIPTION = "C++ header only log"
```
Comment (andr2000) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xt-log/xt-log_git.bb]: I think this should be split: common goes to meta-xt-images and the rest stays here as bbappend

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-core/dbus/dbus-cxx_0.10.0.bb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+DESCRIPTION = "C++ bindings for dbus"
```
Comment (andr2000) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-core/dbus/dbus-cxx_0.10.0.bb]: This should be split: meta-xt-images + bbappend

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/kernel-module-gles/kernel-module-gles_1.10.bbappend [hunk 3]
```diff
@@ -1,5 +1,6 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
```
Comment (andr2000) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/kernel-module-gles/kernel-module-gles_1.10.bbappend]: why is this needed?

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/gles-module/gles-user-module_1.10.bbappend [hunk 4]
```diff
@@ -1,4 +1,5 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
```
Comment (andr2000) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/gles-module/gles-user-module_1.10.bbappend]: why is this needed?

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/xendriverdomain.service [hunk 5]
```diff
@@ -0,0 +1,12 @@
+[Unit]
```
Comment (andr2000) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/xendriverdomain.service]: Why is that we need to craft this file ourselves now? how was it implemented w/o this file before thud?

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/doc/local.conf.rcar-domd-image-weston [hunk 6]
```diff
@@ -391,3 +391,5 @@ MACHINEOVERRIDES .= ":rcar"
 DISTRO_FEATURES_remove = " x11 gtk gobject-introspection-data nfc irda zeroconf 3g"
 ASSUME_PROVIDED += "sync-native"
 HOSTTOOLS += "sync"
+ASSUME_PROVIDED += "bison-native"
```
Comment (andr2000) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/doc/local.conf.rcar-domd-image-weston]: why? can't Yocto build bison by itself?

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/wayland/wayland-ivi-extension_2.0.0.bbappend [hunk 7]
```diff
@@ -1,2 +1,2 @@
 SRC_URI = "git://github.com/GENIVI/${BPN}.git;protocol=http"
-SRCREV = "2.2.0"
+SRCREV = "2.2.0_weston_5_compartible"
```
Comment (andr2000) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/wayland/wayland-ivi-extension_2.0.0.bbappend]: just curious why this specific branch?

File: recipes-domd/domd-image-weston/domd-image-weston.bbappend [hunk 8]
```diff
@@ -75,7 +75,6 @@ python do_domd_install_machine_overrides() {
 
 XT_QUIRK_PATCH_SRC_URI_rcar = "\
     file://${S}/meta-renesas/meta-rcar-gen3/docs/sample/patch/patch-for-linaro-gcc/0001-rcar-gen3-add-readme-for-building-with-Linaro-Gcc.patch;patchdir=meta-renesas \
-    file://0001-hack-Remove-dependency-to-multi-user.target-from-wes.patch;patchdir=meta-renesas \
```
Comment (andr2000) [inline recipes-domd/domd-image-weston/domd-image-weston.bbappend]: I see you **remove** the patch, but I can't find where it is **moved** as the title says

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-multimedia/pulseaudio/pulseaudio_%.bbappend [hunk 9]
```diff
@@ -0,0 +1,7 @@
+do_install_append () {
```
Comment (andr2000) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-multimedia/pulseaudio/pulseaudio_%.bbappend]: is this considered as hack? should we put a comment on this why we need to behave this way?

File: recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend [hunk 10]
```diff
@@ -22,7 +22,7 @@ python __anonymous () {
 ################################################################################
 # Generic ARMv8
 ################################################################################
-SRC_URI = "repo://github.com/xen-troops/manifests;protocol=https;branch=master;manifest=prod_devel/dom0.xml;scmdata=keep"
+SRC_URI = "repo://github.com/iartemenko/manifests;protocol=https;branch=pr_prod_thud;manifest=prod_devel/dom0.xml;scmdata=keep"
```
Comment (andr2000) [inline recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend]: this patch must be removed?

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xt-log/xt-log_git.bb (hunk 1 above)
Comment (iartemenko) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/xt-log/xt-log_git.bb]: Done

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-core/dbus/dbus-cxx_0.10.0.bb (hunk 2 above)
Comment (iartemenko) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-core/dbus/dbus-cxx_0.10.0.bb]: Done

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/kernel-module-gles/kernel-module-gles_1.10.bbappend (hunk 3 above)
Comment (iartemenko) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/kernel-module-gles/kernel-module-gles_1.10.bbappend]: Removed

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/gles-module/gles-user-module_1.10.bbappend (hunk 4 above)
Comment (iartemenko) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/gles-module/gles-user-module_1.10.bbappend]: Removed

File: recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend (hunk 10 above)
Comment (iartemenko) [reply recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend]: Sure, removed.

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/wayland/wayland-ivi-extension_2.0.0.bbappend (hunk 7 above)
Comment (lorc) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/wayland/wayland-ivi-extension_2.0.0.bbappend]: So, basically we have 3 options:

1. Leave `2.2.0` - then project will not build
2. Use this tag - and we are good to go
3. Use master branch

Taking into account that receipt is named blah-blah-blah_2.0.0, neither option is correct.

Personally, I'm sticking to option 2 + rename receipt to reflect real version.

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/xendriverdomain.service (hunk 5 above)
Comment (iartemenko) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/xendriverdomain.service]: Now it is provided by meta-virtualization layer. So, removing it from guest-addons and let Xen install it.

File: recipes-domd/domd-image-weston/domd-image-weston.bbappend (hunk 8 above)
Comment (iartemenko) [reply recipes-domd/domd-image-weston/domd-image-weston.bbappend]: It was improperly rebased wis 62f05d1. Fixed.

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/wayland/wayland-ivi-extension_2.0.0.bbappend (hunk 7 above)
Comment (iartemenko) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/wayland/wayland-ivi-extension_2.0.0.bbappend]: Done

**Comment by iartemenko** [comment]
Comments addressed

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/doc/local.conf.rcar-domd-image-weston (hunk 6 above)
Comment (andr2000) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/doc/local.conf.rcar-domd-image-weston]: bump

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/wayland/wayland-ivi-extension_2.0.0.bbappend (hunk 7 above)
Comment (andr2000) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/wayland/wayland-ivi-extension_2.0.0.bbappend]: Appropriate comment could be an option here

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-multimedia/pulseaudio/pulseaudio_%.bbappend (hunk 9 above)
Comment (andr2000) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-multimedia/pulseaudio/pulseaudio_%.bbappend]: bump
